src/concrete/ml/pytest/torch_models.py

```diff
@@ -1409,3 +1409,82 @@ def forward(self, x):
         """
 
         return self.sub_module(x)
+
+
+class EncryptedMatrixMultiplicationModel(nn.Module):
+    """PyTorch module for performing matrix multiplication between to encrypted values."""
+
+    # pylint: disable-next=unused-argument
+    def __init__(self, input_output, activation_function):
+        super().__init__()
+        self.act = activation_function()
+
+    def forward(self, input1):
+        """Forward function for matrix multiplication.
+
+        Args:
+            input1 (torch.Tensor): The first input tensor.
+
+        Returns:
+            torch.Tensor: The result of the matrix multiplication.
+        """
+        input1 = input1.unsqueeze(1)
+
+        output = torch.matmul(input1, input1.transpose(2, 1))
+        output = self.act(output)
+
+        # Squeeze out one of the last two singleton dimensions
+        output = output.squeeze(-1)
+        return output
+
+
+class ManualLogisticRegressionTraining(torch.nn.Module):
```

Comment:
is it expected to add training in this PR? it was only for enc GEMM , I don't think it's a good idea to make this PR even more complex

Reply:
This test will be important in the following PRs. Better have it now.

Reply:
ok sure if no one fights for it then good for me